crates/sui-core/src/consensus_validator.rs

```diff
@@ -152,6 +152,10 @@ impl TransactionVerifier for SuiTxValidator {
         self.validate_transactions(txs)
             .map_err(|e| ValidationError::InvalidTransaction(e.to_string()))
     }
+
+    fn vote_batch(&self, _batch: &[&[u8]]) -> Vec<TransactionIndex> {
```

Comment:
not sure yet how you're planning to use this but are you sure return type shouldn't be wrapped in a Result?

Reply:
ok I now see there's a TODO on this interface elsewhere

Reply:
Yes this can be useful. Expanded the TODO.